### ISSUE TITLE
Conduct security review of Teach site

### ISSUE THREAD

**Issue by hannahkane** [mozilla/teach.webmaker.org#680]
Assigning this to Simon as per release checklist in #152

**Comment by hannahkane** [comment]
@simonwex - let me know if you need anything to start this process.

**Comment by simonwex** [comment]
tl;dr: Audited teach-api as well as features of id.webmaker.org used by teach site. Amber-green: _we're good to ship, but we have a little bit of cleanup to do_

 * [ ] Password Complexity
 * [ ] Password Rotation
 * [ ] Account Lockout and Failed Login
 * [ ] Password Reset Functions 
 * [ ] Email Change and Verification Functions 
 * [ ] Password Storage
  * Not storing passwords in id
 * [x] Session ID Creation secure
  * handled by hapi-auth-cookie
 * [ ] Inactivity Time Out
  * minor exposure compared to UX
 * [x] Secure Flag set for cookies 
 * [x] HTTP-Only Flag
 * [ ] Logout
  * SessionID still exists after logout

 * [x] Robust Use of Input Validation
 * [ ] Preventing XSS
  * Require Content Security Policy (CSP)
 * [ ] Preventing SQL Injection
 * [x] CSRF Protection
 * [x] x-frame-options set
 * [x] 3rd Party Scripts
 * [x] Strict-Transport-Security set
 * [x] Error Handling audited for sensitive information in User Facing Error Messages

**Comment by hannahkane** [comment]
Just to confirm, we're good to launch *before* fixing these things, but we need to file tickets to fix them soon?

**Comment by hannahkane** [comment]
@simonwex and @toolness - I'd like to close this ticket, but want to make sure we've accounted for the unchecked boxes. Can you guys file any relevant tickets, please?

**Comment by simonwex** [comment]
Good to close. The important ones have been filed separately.
